product_weight_through_uom/models/product_template.py

```diff
@@ -19,6 +21,21 @@ class ProductTemplate(models.Model):
         default=False,
     )
 
+    @api.depends('product_variant_ids', 'product_variant_ids.weight')
+    def _compute_extra_weight(self):
+        unique_variants = self.filtered(
+            lambda x: len(x.product_variant_ids) == 1)
+        for template in unique_variants:
+            template.extra_weight = template.product_variant_ids.extra_weight
+        for template in (self - unique_variants):
```

Comment:
No need of this. ORM already initializes to 0 all values before calling compute method. Any way, a better alternative would be:
```python
(self - unique_variants).update({'extra_weight': 0.0})
```